protostar/composition_root.py

```diff
@@ -62,6 +66,13 @@ def build_di_container(script_root: Path, start_time: float = 0):
         protostar_toml_path.parent if protostar_toml_path is not None else cwd
     )
     protostar_toml_path = protostar_toml_path or project_root_path / "protostar.toml"
+
+    command_names_delayed_provider = CommandNamesDelayedProvider()
```

Comment:
why is that class used here? it literally checks if the provider is not `None` - and in which case exactly `protostar_cli` is expected to be `None`?

Reply:
I think what @Arcticae meant is why do we hoist it as DI if we just hard code it? I agree with him we shouldn't do that

Reply:
I am not sure if I get it. Should `set_command_names_provider` be part of `ConfigurationFile` interface and that class removed?

Reply:
We should not inject it as a dependency